Makefile.base

```diff
@@ -67,9 +67,11 @@ ifeq ($(strip $(ASSMSRC))$(NO_AUTO_SRC),)
   ASSMSRC := $(wildcard *.S)
 endif
 
-ifneq (,$(SRCXX))
-  ifeq (,$(filter cpp,$(FEATURES_USED)))
-    $(error Found C++ source, but feature "cpp" is not used. Add "FEATURES_REQUIRED += cpp")
+ifneq (1,$(TEST_KCONFIG))
+  ifneq (,$(SRCXX))
+    ifeq (,$(filter cpp,$(FEATURES_USED)))
+      $(error Found C++ source, but feature "cpp" is not used. Add "FEATURES_REQUIRED += cpp")
```

Comment:
How is this checked when using Kconfig?

Reply:
Hmmm true, can kconfig be aware of this? or can we apply a check after kconfig resolved the features?

Reply:
Turns out it was pretty easy.  I tested by changing helloworld to cpp.  I also exposed the CPP and cppstdlib modules as it does make sense to select them.